File: base/src/main/vsdk/toolkit/common/VSDKFatalException.java
package vsdk.toolkit.common;
import java.io.Serial;

/**
Unchecked exception raised when VSDK reports a fatal error in non-exit mode.
*/
public class VSDKFatalException extends RuntimeException {
    @Serial private static final long serialVersionUID = 20260417L;

    public VSDKFatalException(String message) {
        super(message);
    }

    public VSDKFatalException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: typescript/base/src/main/vsdk/toolkit/common/VSDKFatalException.ts
import { RuntimeException } from "../../../java/lang/RuntimeException.js";
import { Throwable } from "../../../java/lang/Throwable.js";

/** Unchecked failure reported by VSDK when fatal process exit is disabled. */
export class VSDKFatalException extends RuntimeException {
  public constructor(message: string, cause?: Throwable) {
    super(message, cause);
    this.name = "VSDKFatalException";
  }
}
